sale_order_weight/models/sale_order.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 Manuel Calero Solís (https://xtendoo.es)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    total_ordered_weight = fields.Float(
+        compute="_compute_total_ordered_weight",
+        string="Ordered Weight",
+        store=True,
+    )
+
+    @api.depends("order_line.total_ordered_weight")
+    def _compute_total_ordered_weight(self):
+        for order in self:
```

Comment:
this field can be used in tree views too. It would be better to use a read_group + sum function

Reply:
Hi. It is a stored field. so the function will not be call by a tree view displaying 10000 sale.order.line. (for exemple).
or did I missed something ?